### PR TITLE
Basic CLI Implementation 2

### PR DESCRIPTION
**Pull request by MaxIsWell42**

### CODE REVIEW

**Review by sethitow — CHANGES_REQUESTED** [review]
Please remove `.DS_Store`, `stc`, `main`, and `example` binary files from source control.

Because a new library (Boost) is used in your changes, you’ll need to edit `ci.yml` in order for builds to pass. This will involve adding a new build step that runs `apt-get install libboost-all-dev`. You’ll also need to lower the minimum required Boost version in CMakeLists.txt

**Comment by MaxIsWell42** [comment]
> Please remove `.DS_Store`, `stc`, `main`, and `example` binary files from source control.
> 
> Because a new library (Boost) is used in your changes, you’ll need to edit `ci.yml` in order for builds to pass. This will involve adding a new build step that runs `apt-get install libboost-all-dev`. You’ll also need to lower the minimum required Boost version in CMakeLists.txt

Made these changes. Set boost version to need to be at least 1.7.0 instead of 1.7.6, but still can't find .DS_Store even when  viewing hidden files.

File: src/compile.hpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#ifndef COMPILE_HPP
+#define COMPILE_HPP
+
+#include <iostream>
```
Comment (sethitow) [inline src/compile.hpp]: `iostream` doesn’t need to be included here
```suggestion
```

File: src/parse.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#ifndef PARSE_HPP
+#define PARSE_HPP
+
+#include <iostream>
```
Comment (sethitow) [inline src/parse.hpp]: `iostream` doesn’t need to be included here
```suggestion
```

**Review by sethitow — CHANGES_REQUESTED** [review]
Please squash commits, and I think we’re good to merge!

**Review by sethitow — APPROVED** [review]

**Pull request merged**